### DISCUSSION TITLE
Roadmap Discussion for JSON Schema – Community Input Needed

### DISCUSSION THREAD

**Discussion by benjagm** [json-schema-org/community#813] [Vision and direction]
**Objective:**  
We’d like to open an issue to invite everyone in the JSON Schema Community to share their thoughts and proposals on our priorities for the next two years. The time for us is exciting, but it is important to have this discussion to make sure we invest our efforts in what is best fort JSON Schema Specification and the JSON Schema Community.

We encourage participation from everyone: users, contributors, implementers, adopters, sponsors, ambassadors, and the TSC. Your feedback and involvement are crucial in shaping our future direction.

### Next Steps

Please share your thoughts as comments on this discussion. We want to hear your ideas, feedback, and suggestions!

**How to Contribute:**  
- Comment below with your insights or proposals.
- If you have specific projects or initiatives in mind, please share those as well.
- Feel free to tag others in the community who should be part of this discussion.

Let’s work together to create a roadmap that reflects the needs and aspirations of our community. Your participation is crucial!

Thank you, and We look forward to our collective brainstorming!

**Comment by benjagm** [comment]
My proposal of focus areas are:

1. **Specification**  
   - Stable Spec development.
   - Spec Development and publication lifecycle.

2. **Open Governance**  
   - Strengthen our governance model to ensure transparency and inclusivity in decision-making processes.
   - Make it easier for contributors to join governing roles.

3. **Support for Implementers**  

4. **Ecosystem Collaboration**  
   - Foster collaborations with related ecosystems, such as OpenAPI.

5. **Sponsorship Retention and Attraction**  

6. **JSON Schema Conference**  

7. **Join an Open Source Foundation**

**Reply by gregsdennis** [reply]
I think (7) can probably help and inform a lot of this, especially (2) and (6).

**Comment by jviotti** [comment]
@benjagm 's proposal looks really good. What I would specifically highlight apart from that, is that we should double down on how we market ourselves. I want to fix what many others raised before:

- Lots of companies/developers are not even aware they are using JSON Schema because they thing its just OpenAPI, etc
- Lots of companies/developers think JSON Schema is just the configuration format of AJV
- As a consequence, we a LOT less attention that higher level projects that hugely depend on us. We should be making big fuss about who we are and the value we bring to the table

We need a plan to properly educate people and tackle these common confusions, otherwise everything else we do has less impact.

**Comment by AymarN** [comment]
[Insight] Environmentally friendly applications.

The Json Schema project requires Node.js v20.9.0+ like many updated applications on the market.
All computers are not aligned with that version of Node.js. And the situation grows grave on Linux distributions
such as Centos. There is a limitation in regard to the processor used.

A solution might be to run those applications on Docker and check that all computers retrieve the specification
An alternative to this issue is also to archive those applications versions.  Node.js v20.9.0+ is not aligned with all
computers however Node.js v8 does . Several editors are not provided the previous versions of their applications.

[Proposal] 
Archive the previous versions of the specification under Node.js v16
An official image of that specification on Docker.
Thank you to let us share our thoughts.

**Reply by gregsdennis** [reply]
Hey @AymarN.  I'm not sure I understand.  JSON Schema is only a specification; it doesn't "run".

Are you referencing a specific implementation? If so, this suggestion would be good for the implementation, but we don't control that here.

**Reply by AymarN** [reply]
Thank you for your feedback. 

**_The set up utilizes Node.js v20.9.0+._** There are several applications which were upgraded to Node.js v20.9.0+.
I do not have any issues to set up the full documentation on Windows using an intel :Intel(R) Core(TM) i5-5200U 
CPU @ 2.20GHz   2.20 GHz

The full documentation may operate however not on all computers particularly on Linux  distributions such as 
 Centos 7. There are limitations on some processors to install Node.js >= V16. 

A proposal could be to archive the previous specifications that operates under Node.js < V16. I do not 
comprehend how they were implemented before and which version of node was used at that moment
for the previous drafts. May you help ?

I made also a second proposal.

**Reply by gregsdennis** [reply]
> There are several applications which were upgraded to Node.js v20.9.0+

I think this is where I'm confused.  The only thing controlled by JSON Schema that could remotely be considered an "application" is the website, and it's just static content.  It has JS that runs in the browser, but that doesn't require Node.

To what applications are you referring, and where is Node being used?

**Reply by jdesrosiers** [reply]
This thread is off topic for this discussion. The roadmap is for big picture goals and objectives. This is an operational detail. Please create a new issue/discussion if you want to continue the discussion about Node.js versions.
